sympy/core/evalf.py

```diff
@@ -82,7 +93,13 @@ class PrecisionExhausted(ArithmeticError):
 """
 
 
-def fastlog(x):
+MPF_VALUE_TUPLE = tTuple[int, int, int, int]
+SCALED_ZERO_TUPLE = tTuple[List[int], int, int, int]
+TEMP_RESULT = Any
+# tUnion[tTuple[Optional[MPF_VALUE_TUPLE], Optional[MPF_VALUE_TUPLE], Optional[int], Optional[int]], 'ComplexInfinity']
+
+
+def fastlog(x: Optional[MPF_VALUE_TUPLE]) -> tUnion[int, Any]:
```

Comment:
Why is this `Optional`?

Reply:
Because the body of `fastlog` expect this case.
https://github.com/sympy/sympy/blob/9019765edb0d0b4bbd29674a58b93ef4e75b92f8/sympy/core/evalf.py#L117

Reply:
I guess what I mean is when is this called with `None`?

Reply:
https://github.com/sympy/sympy/blob/d04fa46664c3aeda08fb41a2b2d62055056e8926/sympy/core/evalf.py#L880-L881
or
https://github.com/sympy/sympy/blob/d04fa46664c3aeda08fb41a2b2d62055056e8926/sympy/core/evalf.py#L1000-L1006
where the argument of `fastlog` is directly from a call to `evalf`s without checking `None`.